### ISSUE TITLE
Unable to deploy AWS budget dashboard - Column 'collection_time' cannot be resolved

### ISSUE THREAD

**Issue by work-bw** [aws-samples/aws-cudos-framework-deployment#986]
Trying to deploy the Budgets dashboard via cid-cmd deploy.  I need some help resolving the installation issue please.  Thanks.

Brian
```
=======================

? [dashboard-id] Please select a dashboard to deploy:   [aws-budgets] AWS Budgets
Dashboard "aws-budgets" is not deployed

Required datasets: 
 - budgets_view


Looking by DataSetId defined in template...complete

There are still 1 datasets missing: budgets_view
Creating dataset: budgets_view

? [athena-workgroup] Select Amazon Athena workgroup to use: CID

? [quicksight-datasource-id] Please choose DataSource (Select the first one if not sure): CID-CMD-Athena <CREATE NEW DATASOURCE>

? [quicksight-datasource-role] Please choose a QuickSight role. It must have access to Athena: CidCmdQuickSightDataSourceRole

? [athena-database] Select AWS Athena database to use: athenacurcfn_nelnet_cudos
Role CidCmdQuickSightDataSourceRole was updated. https://console.aws.amazon.com/iam/home?#/roles/details/CidCmdQuickSightDataSourceRole
ERROR - Data source already exists {'Arn': 'arn:aws:quicksight:us-west-2:704691252361:datasource/CID-CMD-Athena', 'DataSourceId': 'CID-CMD-Athena', 'Name': 'CID-CMD-Athena', 'Type': 'ATHENA', 'Status': 'CREATION_SUCCESSFUL', 'CreatedTime': datetime.datetime(2023, 7, 3, 19, 3, 21, 937000, tzinfo=tzlocal()), 'LastUpdatedTime': datetime.datetime(2023, 7, 3, 19, 3, 23, 202000, tzinfo=tzlocal()), 'DataSourceParameters': {'AthenaParameters': {'WorkGroup': 'primary'}}, 'SslProperties': {'DisableSsl': False}}
athena_datasource <cid.helpers.quicksight.datasource.Datasource object at 0x7fafd68835e0>
Detected views: account_map
Missing views: budgets_view
CRITICAL - Athena query failed: line 12:3: Column 'collection_time' cannot be resolved

```
====================

When I'm prompted to select the quicksight-datasource-id my only choice is to create new datasource.  It doesn't seem to see the existing datasources.
```
? [quicksight-datasource-id] Please choose DataSource (Select the first one if not sure): (Use arrow keys)
 » CID-CMD-Athena <CREATE NEW DATASOURCE>

```

**Comment by work-bw** [comment]
? [quicksight-datasource-id] Please choose DataSource (Select the first one if not sure): (Use arrow keys)
 » CID-CMD-Athena <CREATE NEW DATASOURCE>

**Comment by work-bw** [comment]
`It didn't see to show the CID-CMD-Athena <CREATE NEW DATASOURCE> line`

**Comment by work-bw** [comment]
In the last line of the original comment

**Comment by iakov-aws** [comment]
Hello, 

Collection time must be there in table : https://github.com/awslabs/cid-framework/blob/main/data-collection/deploy/module-budgets.yaml#L165

Can you check the version of data collection (in description of the stack) and update to the latest https://catalog.workshops.aws/awscid/en-US/data-collection/update

**Comment by iakov-aws** [comment]
Brian @work-bw 
Regarding datasources, can you check if datasource status are ok?

some what like this in cloud shell:
```
AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query 'Account' --output text)
aws quicksight list-data-sources  --aws-account-id $AWS_ACCOUNT_ID --query 'DataSources[?Type==`ATHENA`]'
```

**Comment by work-bw** [comment]
Can you let me know the status of this issue please or the next steps?  I tried checking the latest cid-cmd updates and I still get the same error trying to deploy the TAO or Budget dashboards.  Is this an issue you are able to recreate?  Is there other data I can provide to help troubleshoot?  Thank you.

Brian

**Comment by iakov-aws** [comment]
Hi Brain, 
I contacted your TAM. I suggest one of our team members to have a call with you to make sure all issues got fixed.

**Comment by VoicuAWS** [comment]
I see you selected the workgroup CID but none of the datasources are created under workgroups CID, that is why none show up. The existing "CID-CMD-Athena" datasource is under "primary" workgroup per your output above, this is why it is prompting you to create one. If you select primary in the workgroup selection step, it should pickup exiting "CID-CMD-Athena" datasource.

What I am not sure though is why the creation of the new datasource fails. When we try to create it it should use the "CID" workgroup, but the error we see in your log says it is using "primary". We need to understand why that happens. Are you using the latest cid-cmd tool? Logs would help here, but we can discuss more in our upcoming call. 
I tried to repro but I did not get the error.
